app/domain/authentication/authn_jwt/jwt_fetch_provider_uri_signing_key.rb

```diff
@@ -0,0 +1,55 @@
+module Authentication
+  module AuthnJwt
+
+    JwtFetchProviderUriSigningKey = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger,
+        discover_identity_provider: OAuth::DiscoverIdentityProvider.new,
+        fetch_secrets: Conjur::FetchRequiredSecrets.new
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      def call
+        provider_uri
```

Comment:
you don't need to have this method here. It memoizes an object so it doesn't need to be initialized. you can just call it when you need it.